dev/ToastNotifications/ToastNotificationManager.cpp

```diff
@@ -124,14 +126,41 @@ namespace winrt::Microsoft::Windows::ToastNotifications::implementation
         THROW_HR_IF(E_UNEXPECTED, notificationId == 0);
         toast.ToastId(notificationId);
     }
-    winrt::Windows::Foundation::IAsyncOperation<winrt::Microsoft::Windows::ToastNotifications::ToastProgressResult> ToastNotificationManager::UpdateToastProgressDataAsync(winrt::Microsoft::Windows::ToastNotifications::ToastProgressData /* data */, hstring /* tag */, hstring /* group */)
+
+    // API updates notification data when both tag and group are provided or when tag is provided.
+    // For other scenarios the API throws
+    winrt::Windows::Foundation::IAsyncOperation<winrt::Microsoft::Windows::ToastNotifications::ToastProgressResult> ToastNotificationManager::UpdateToastProgressDataAsync(winrt::Microsoft::Windows::ToastNotifications::ToastProgressData const& data, hstring const& tag, hstring const& group)
     {
-        throw hresult_not_implemented();
+        if ((tag == winrt::hstring(L"") && group == winrt::hstring(L"")) || (tag == winrt::hstring(L"") && group != winrt::hstring(L"")))
+        {
+            co_return winrt::ToastProgressResult::NotificationNotFound;
+        }
+
+        std::wstring appId{ RetrieveToastAppId() };
+
+        winrt::com_ptr<::ABI::Microsoft::Internal::ToastNotifications::IToastProgressData> toastProgressData{ winrt::make_self<ToastProgressDataABI>(data) };
+
+        auto hr = ToastNotifications_UpdateNotificationData(appId.c_str(), tag.c_str(), group.c_str(), toastProgressData.get());
+
+        THROW_HR_IF(hr, (hr != S_OK) && (hr != E_NOT_SET));
+
+        if (SUCCEEDED(hr))
+        {
+            co_return winrt::ToastProgressResult::Succeeded;
+        }
+        else if (hr == E_NOT_SET)
+        {
+            co_return winrt::ToastProgressResult::NotificationNotFound;
+        }
     }
-    winrt::Windows::Foundation::IAsyncOperation<winrt::Microsoft::Windows::ToastNotifications::ToastProgressResult> ToastNotificationManager::UpdateToastProgressDataAsync(winrt::Microsoft::Windows::ToastNotifications::ToastProgressData /* data */, hstring /* tag */)
+
+    winrt::Windows::Foundation::IAsyncOperation<winrt::Microsoft::Windows::ToastNotifications::ToastProgressResult> ToastNotificationManager::UpdateToastProgressDataAsync(winrt::Microsoft::Windows::ToastNotifications::ToastProgressData const& data, hstring const& tag)
     {
-        throw hresult_not_implemented();
+        auto toastProgressResult = co_await UpdateToastProgressDataAsync(data, tag, L"");
```

Comment:
Can you directly return `UpdateToastProgressDataAsync` in the `co_return`?

Reply:
No it doesn't like it that way.